File: packages/depz-sensor-sdk-java/src/main/java/ai/depz/sensor/transport/CrcType.java
package ai.depz.sensor.transport;

/** Payload CRC type carried in the two top bits of the frame data-size field. */
public enum CrcType {
    NONE(0, 0),
    CRC8(1, 1),
    CRC16(2, 2),
    CRC32(3, 4);

    public final int value;
    public final int size;

    CrcType(int value, int size) {
        this.value = value;
        this.size = size;
    }

    public static CrcType fromValue(int value) {
        for (CrcType t : values()) {
            if (t.value == value) {
                return t;
            }
        }
        throw new IllegalArgumentException("invalid crc_type: " + value);
    }
}

File: packages/depz-sensor-sdk-csharp/src/Depz.Sensor/Transport/CrcType.cs
namespace Depz.Sensor.Transport;

/// <summary>Payload CRC type carried in the two high bits of the data-size field.</summary>
public enum CrcType
{
    None = 0,
    Crc8 = 1,
    Crc16 = 2,
    Crc32 = 3,
}

public static class CrcTypeExtensions
{
    /// <summary>Size in bytes of the CRC trailer for a non-empty payload.</summary>
    public static int Size(this CrcType crcType) => crcType switch
    {
        CrcType.None => 0,
        CrcType.Crc8 => 1,
        CrcType.Crc16 => 2,
        CrcType.Crc32 => 4,
        _ => throw new ArgumentOutOfRangeException(nameof(crcType)),
    };
}
